modules/csharp/pages/working-with-vector-search.adoc

```diff
@@ -0,0 +1,227 @@
+|
+
+|===
+
+CAUTION: Altering the default training sizes could be detrimental to the accuracy of returned results produced by the model and total computation time.
+
+== Generating Vectors
+
+You can use two methods to generate vectors in Couchbase Lite:
+
+. You can call a Machine Learning(ML) model, and embed the generated vectors inside the documents.
+
+. You can use the `prediction()` function to generate vectors to be indexed for each document at the indexing time.
+
+Below are example configurations of the previously mentioned methods.
+
+=== Create a Vector Index with Embeddings
+
+This method shows you how to create a Vector Index with embeddings.
+
+[source, csharp]
+----
+
+include::csharp:example$code_snippets/VectorSearch.cs[tags=vs-create-index]
+
+----
+
+. First, create the standard configuration, setting up an expressions, number of dimensions and number of centroids for the vector embedding.
+
+. Next, create a collection variable passing it the contents of a given collection from our `database` object, in our case, the `colors` collection.
+
+. Finally, create a vector index, `colors_index`, on a collection and pass it our configuration.
+
+=== Create Vector Index Embeddings from a Predictive Model
+
+This method generates vectors to be indexed for each document at the index time by using the `prediction()` function.
+The key difference to note is that the `config` object uses the output of the `prediction()` function as the `expression` parameter to generate the color embedding.
+
+[source, csharp]
+----
+        
+include::csharp:example$code_snippets/VectorSearch.cs[tags=vs-create-predictive-index]
+
+----
+
+NOTE: You can use less storage by using the `prediction()` function as the encoded vectors will only be stored in the index. 
+However, the index time will be longer as vector embedding generation is occurring at run time.
+
+== Vector SQL++ Functions
+
+Couchbase Lite currently supports two SQL++ functions, `vector_match()` and `vector_distance()`.
+
+=== `vector_match(vectorIndexIdentifier, targetVectorExpr, [limit = 3])`
+
+[cols = "3*"]
+|===
+|Parameter |Is Required |Description
+
+|vectorIndexIdentifier
+|Required
+|The name of the vector index to perform the vector search on.
+|targetVectorExpr
+|Required
+|The target vector expression that returns a vector in the form of an array of numbers.
+|limit
+|Optional
+|The limit number of the returned matched results.
+The maximum number allowed is 10000. An error will be returned when creating a query with a limit greater than 10000.
+
+|===
+
+NOTE: The default value for the `limit` parameter is 3.
+
+=== Use `vector_match()`
+
+[source, csharp]
+----
+        
+include::csharp:example$code_snippets/VectorSearch.cs[tags=vs-use-vector-match]
+
+----
+
+This function performs vector search against a specific vector index identifier for the specified vector expression.
+If the specified index does not exist, an error will occur on creation of the query.
```

Comment:
If the specified index does not exist, creating the query results in an error.

(I tried to think of how to reword this somewhere above, just kind of got it now. 
Try to minimize the passive/future even though it's wicked hard)